packages/tsconfig-reference/copy/en/options/disableSourceOfProjectReferenceRedirect.md

```diff
@@ -1,7 +1,49 @@
 ---
-display: "Disable Source Project Reference Redirect"
-oneline: "Disable preferring source files instead of declaration files when referencing composite projects."
+display: "Disable Redirections to Project Reference Source Files"
+oneline: "When editing code, only consult declaration files from referenced projects."
 ---
 
-When working with [composite TypeScript projects](/docs/handbook/project-references.html), this option provides a way to go [back to the pre-3.7](/docs/handbook/release-notes/typescript-3-7.html#build-free-editing-with-project-references) behavior where d.ts files were used to as the boundaries between modules.
-In 3.7 the source of truth is now your TypeScript files.
+When editing a project with [project references](/docs/handbook/project-references.html), the TypeScript language service does not require us to build all dependencies so that generated declaration files are available.
+This is subtle, because this behavior means things usually "just work" in the editor, but differs from how project reference builds work, and warrants explanation.
+
+When building project references, the compiler always expects declaration files to be generated so that less code can be held in memory at once.
+This behavior works for performing a full build, but is often confusing when editing code.
+When referencing code from another project, most people expect the changes to be available immediately in the editor.
+
+For example, imagine making the following change to `projectB/src/setColor.ts`
+
+```diff ts
+  export Options {
+      hue: number;
+      saturation: number;
+-     value: number;
++     lightness: number;
+  }
+
+  export function setColor(options: Options): void {
+      // ...
+  }
+```
+
+If we want to update `projectA/src/draw.ts` to call `setColor` with `lightness` instead of `value` as so:
+
+```diff ts
+  function draw() {
+      // ...
+      setColor({
+          hue: getUserSetting("hue"),
+          saturation: getUserSetting("saturation"),
+-         value: getUserSetting("value"),
++         lightness: getUserSetting("lightness"),
+      })
+  }
+```
+
+we don't want to stop what we're doing and build `projectB` before we can get accurate errors in `projectA`, so TypeScript prefers reading source `.ts` files instead of generated `.d.ts` files if they're around.
+
```

Comment:
we always get source file even if project is not yet built and hence no d.ts file